common/Services/NotificationService.cs

```diff
@@ -0,0 +1,214 @@
+
+    private readonly IWindowsIdentityService _windowsIdentityService;
+
+    private readonly string _componentName = "NotificationService";
+
+    private readonly StringResource _stringResource;
+
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    private StackedNotificationsBehavior? _notificationQueue;
+
+    public NotificationService(IWindowsIdentityService windowsIdentityService)
+    {
+        _windowsIdentityService = windowsIdentityService;
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+        _stringResource = new StringResource("DevHome.Common.pri", "DevHome.Common/Resources");
+    }
+
+    public bool ShowHyperVAdminWarningToast()
+    {
+        // Temporary toast notification to inform the user that they are not in the Hyper-V admin group.
+        // In the future we'll use an admin process from Dev Home to add the user to the group.
+        var toast = new AppNotificationBuilder()
+           .AddText("Warning")
+           .AddText(StringResourceHelper.GetResource(StringResourceHelper.UserNotInHyperAdminGroupMessage))
+           .AddButton(new AppNotificationButton(StringResourceHelper.GetResource(StringResourceHelper.UserNotInHyperAdminGroupButton))
+           .AddArgument("action", "AddUserToHyperVAdminGroup"))
+           .BuildNotification();
+
+        AppNotificationManager.Default.Show(toast);
+        return toast.Id != 0;
+    }
+
+    public void Initialize(StackedNotificationsBehavior notificationQueue)
+    {
+        _notificationQueue = notificationQueue;
+    }
+
+    public void HandlerNotificationActions(AppActivationArguments args)
+    {
+        if (args.Data is ToastNotificationActivatedEventArgs toastArgs)
+        {
+            try
+            {
+                if (toastArgs.Argument.Contains("action=AddUserToHyperVAdminGroup"))
+                {
+                    // Launch compmgmt.msc in powershell
+                    var psi = new ProcessStartInfo();
+                    psi.FileName = "powershell";
+                    psi.Arguments = "Start-Process compmgmt.msc -Verb RunAs";
+                    Process.Start(psi);
+                }
+            }
+            catch (Exception ex)
+            {
+                _log.Error(_componentName, $"Unable to launch computer management due to exception", ex);
+            }
+        }
+    }
+
+    public void ShowRestartNotification()
+    {
+        if (_notificationQueue != null)
+        {
+            var command = new RelayCommand(() =>
+            {
+                System.Diagnostics.Process process = new System.Diagnostics.Process();
+                System.Diagnostics.ProcessStartInfo startInfo = new System.Diagnostics.ProcessStartInfo();
+                startInfo.WindowStyle = System.Diagnostics.ProcessWindowStyle.Hidden;
+                startInfo.FileName = "cmd.exe";
```

Comment:
Do you need to call cmd.exe to call shutdown.exe? Can't you call shutdown.exe directly?

Reply:
Didn't think of this. Changed.